platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateRoot.java

```diff
@@ -365,7 +375,24 @@ public void preHandle(@NonNull final Event event) {
      * {@inheritDoc}
      */
     @Override
-    public MerkleNode migrate(final int ignored) {
+    public MerkleNode migrate(final int version) {
+        if (version < VERSION_2) {
+            final var zerothChild = getChild(0);
+            if (!(zerothChild instanceof PlatformState platformState)) {
+                throw new IllegalStateException("Expected a PlatformState as the first child");
+            }
+            preV054PlatformState = platformState;
```

Comment:
Which `platformState` is it at line 384? It cannot be the one declared at line 381 because the latter is already out of scope here. It appears that you're reading the value of the class field instead of the zerothChild here, which may not be what you want to read here.

See https://blogs.oracle.com/javamagazine/post/java-pattern-matching-instanceof-flow-scoping

Reply:
It's actually `platformState` for that `if` statement
Take a look at this example from the link you provided
![image](https://github.com/user-attachments/assets/3c8c27b0-4d72-418c-b18e-ed3266ff572d)

Reply:
Oh, I missed this part. Interesting. It doesn't look super intuitive though, so myself, I'd prefer to avoid this particular version of the pattern. But it's good to know there's not a bug.